modules/library/render/src/main/java/org/geotools/renderer/style/FontCache.java

```diff
@@ -252,4 +259,31 @@ public synchronized void resetCache() {
 
         return availableFonts;
     }
+
+    /**
+     * Given a font name, returns alternatives for other scripts, based on the assumption they start
+     * with the same base name.
+     *
+     * @param name
+     * @return A list of font names with the same base name
+     */
+    public List<String> getAlternatives(String name) {
+        List<String> result = alternatives.get(name);
+        if (result == null) {
+            result =
+                    FontCache.getDefaultInstance()
+                            .getAvailableFonts()
+                            .stream()
+                            .filter(f -> f.startsWith(name))
+                            .filter(
+                                    f -> { // leave out alterations, use base fonts
+                                        String lc = f.toLowerCase();
+                                        return !lc.contains(" bold") && !lc.contains(" italic");
+                                    })
+                            .sorted() // leave further altered fonts down the line, base ones first
+                            .collect(Collectors.toList());
+        }
```

Comment:
Should this put the result into alternatives before returning? As far as I can tell, nothing ever gets written to alternatives currently.

Reply:
Good catch, thanks!